### ISSUE TITLE
how to use the model i have trained in other net if i just want to extract the position and size of bbox?

### ISSUE THREAD

**Issue by qin79** [eriklindernoren/PyTorch-YOLOv3#594]
expect to be fast and accurate,which file (detect.py/test.py)should i use and revise ?
code:
 # Initiate model
    model = Darknet(opt.model_def).to(device)
    if opt.weights_path.endswith(".weights"):
        # Load darknet weights
        model.load_darknet_weights(opt.weights_path)
    else:
        # Load checkpoint weights
        model.load_state_dict(torch.load(opt.weights_path))

i trained on coco and save ".pth".but i dont know which one to load

**Comment by qin79** [comment]
can anybody teach me how to revise the code to get the information of bbox?555

**Comment by Flova** [comment]
Currently, I would recommend modifying the `detect,py` to your needs at the moment. I will modify it later to import it as a python module.

Does this answer your question?

**Comment by qin79** [comment]
> Currently, I would recommend modifying the `detect,py` to your needs at the moment. I will modify it later to import it as a python module.
> 
> Does this answer your question?

i agree with you too. i have changed the code (line113 in detect.py),but it still doesn't work. i'm not sure if box_w/box_h represent the real size?
            for x1, y1, x2, y2, conf, cls_conf, cls_pred in detections:

                print("\t+ Label: %s, Conf: %.5f" % (classes[int(cls_pred)], cls_conf.item()))

                box_w = x2 - x1
                box_h = y2 - y1
                box_size = box_w * box_h
                box_mini.extend(box_size, cls_pred)

**Comment by Flova** [comment]
I just tested it and they represent the size in pixel.

``` python
                box_w = x2 - x1
                box_h = y2 - y1

                print([int(x) for x in [box_w, box_h, box_w*box_h]])
```

Adding this code to the current master version and running it on the sample images results in the following printout

```
Saving images:
(0) Image: 'data/samples/dog.jpg'
	+ Label: dog, Conf: 0.99335
[184, 300, 55428]
	+ Label: bicycle, Conf: 0.99981
[425, 333, 141621]
	+ Label: truck, Conf: 0.94229
[220, 87, 19245]
(1) Image: 'data/samples/eagle.jpg'
	+ Label: bird, Conf: 0.99703
[527, 371, 195780]
(2) Image: 'data/samples/field.jpg'
	+ Label: person, Conf: 0.99996
[86, 291, 25275]
	+ Label: horse, Conf: 0.99977
[207, 218, 45429]
	+ Label: dog, Conf: 0.99409
[140, 85, 12086]
(3) Image: 'data/samples/giraffe.jpg'
	+ Label: giraffe, Conf: 0.99959
[290, 371, 107819]
	+ Label: zebra, Conf: 0.97958
[173, 249, 43315]
(4) Image: 'data/samples/herd_of_horses.jpg'
	+ Label: horse, Conf: 0.99459
[286, 217, 62372]
	+ Label: horse, Conf: 0.99352
[211, 179, 37872]
	+ Label: horse, Conf: 0.96845
[157, 137, 21717]
	+ Label: horse, Conf: 0.99478
[152, 195, 29875]
(5) Image: 'data/samples/messi.jpg'
	+ Label: person, Conf: 0.99993
[823, 710, 584624]
	+ Label: person, Conf: 0.99984
[88, 217, 19162]
	+ Label: person, Conf: 0.99996
[494, 613, 303431]
(6) Image: 'data/samples/person.jpg'
	+ Label: person, Conf: 0.99883
[355, 476, 169384]
	+ Label: dog, Conf: 0.99275
[136, 125, 17129]
(7) Image: 'data/samples/room.jpg'
	+ Label: chair, Conf: 0.99906
[79, 110, 8728]
	+ Label: chair, Conf: 0.96942
[68, 131, 8984]
	+ Label: clock, Conf: 0.99971
[16, 15, 260]
(8) Image: 'data/samples/street.jpg'
	+ Label: car, Conf: 0.99977
[99, 43, 4363]
	+ Label: car, Conf: 0.99402
[136, 59, 8128]
	+ Label: car, Conf: 0.99841
[34, 22, 777]
	+ Label: car, Conf: 0.99785
[37, 25, 959]
	+ Label: car, Conf: 0.97907
[79, 35, 2779]
	+ Label: car, Conf: 0.95370
[51, 36, 1889]
	+ Label: traffic light, Conf: 0.99995
[25, 55, 1425]
	+ Label: car, Conf: 0.62254
[72, 55, 4000]
```

**Comment by Flova** [comment]
If you have further questions feel free to reopen the issue.
